core/amber/src/main/python/core/runnables/main_loop.py

```diff
@@ -160,24 +161,32 @@ def process_input_tuple(self) -> None:
                 self.context.tuple_processing_manager.current_input_port_id
             )
 
-        for output_tuple in self.process_tuple_with_udf():
+        for output_data in self.process_tuple_with_udf():
             self._check_and_process_control()
-            if output_tuple is not None:
+            if output_data is not None:
                 self.context.statistics_manager.increase_output_tuple_count(
                     PortIdentity(0)
                 )
                 for to, batch in self.context.output_manager.tuple_to_batch(
-                    output_tuple
+                    output_data
                 ):
                     self._output_queue.put(DataElement(tag=to, payload=batch))
 
+    def process_input_state(self) -> None:
+        self._switch_context()
+        output_state = self.context.marker_processing_manager.get_output_state()
+        self._switch_context()
+        if output_state is not None:
+            for to, batch in self.context.output_manager.emit_marker(output_state):
+                self._output_queue.put(DataElement(tag=to, payload=batch))
+
     def process_tuple_with_udf(self) -> Iterator[Optional[Tuple]]:
         """
         Process the Tuple/InputExhausted with the current link.
 
         This is a wrapper to invoke processing of the executor.
 
-        :return: Iterator[Tuple], iterator of result Tuple(s).
+        :return: Iterator[Tuple], iterator of result Tuple(s) or State.
```

Comment:
your signature states that it returns an Iterator[Optional[Tuple]], but your comments say otherwise. Please update the method name and signature.

Reply:
It will only produce tuples. Fixed.